CMakeLists.txt

```diff
@@ -124,8 +122,21 @@ set(STAGED_INSTALL_PREFIX ${CMAKE_BINARY_DIR}/stage${CMAKE_INSTALL_PREFIX})
 set(MKL_COMPILER_BINDINGS "${CMAKE_CXX_COMPILER_ID}")
 set(BLAS_LANG "CXX")
 set(LAPACK_LANG "CXX")
+#RMR-I modified the ConfigMath scripts so that the first MKL set of libraries it
+# looks for is the ones suggested by this url: 
+#
+# https://software.intel.com/en-us/articles/a-new-linking-model-single-dynamic-library-mkl_rt-since-intel-mkl-103
+#
+# Intel has introduced libmkl_rt.so for exactly the scenario the new inverted
+# Psi4 presents, i.e. a library that is dlopened at runtime.  The point
+# of libmkl_rt.so is that at dlopen time it automatically loads the appropriate
+# mkl libraries instead of having to have rpaths to all possible libraries. We
+# should thus link against it and not try to play Intel link line advisor
+# ourselves.
+#
+# The final link line should be:
+# -L${MKLROOT}/lib/intel64 -lmkl_rt -lpthread -lm -ldl
```

Comment:
Provided I can find a way around it for the purposes of static linking, `mkl_rt.so` os probably a good idea to detect, thanks. Do you happen to know if it can be made to respect sequential/threads via openmp enabling variable?